thorlcr/activities/selfjoin/thselfjoinslave.cpp

```diff
@@ -142,10 +142,15 @@ class SelfJoinSlaveActivity : public CSlaveActivity
         if (!isLocal && TAG_NULL != mpTagRPC)
             sorter.setown(CreateThorSorter(this, server,&container.queryJob().queryIDiskUsage(),&queryJobChannel().queryJobComm(),mpTagRPC));
     }
-    virtual void kill()
+    virtual void kill() override
     {
         sorter.clear();
-        CSlaveActivity::kill();
+        if (portbase)
+        {
+            freePort(portbase, NUMSLAVEPORTS);
+            portbase = 0;
+        }
+        PARENT::kill();
```

Comment:
Does the abort() here need an override ?

Reply:
it doesn't need one..
but all should, I tend to add them when touching a class, so I will add.